src/actions/actions.ts

```diff
@@ -1440,6 +1440,24 @@ class CommandInsertNewLineBefore extends BaseCommand {
 }
 
 @RegisterAction
+class CommandClearLine extends BaseCommand {
+  modes = [ModeName.Normal];
+  keys = ["S"];
+
+  public async exec(position: Position, vimState: VimState): Promise<VimState> {
+    const lineStart = position.getFirstLineNonBlankChar();
```

Comment:
Heh, for a second there I was confused because when I see c-c I normally thing ctrl-c. Took me a sec to realize you meant `cc`!

I'm running Vim 7.4 on OSX and I see the "full line including whitespace" behavior. It's interesting that your Vim has different behavior. In cases like this I think it's best to defer to the manual and remove the whole line.

If you want "change starting at whitespace", there's always `^C`, which is exactly the same number of characters as `cc` :-)